lib/dns/dns.go

```diff
@@ -320,6 +320,48 @@ Redo:
 	return true, nil
 }
 
+// lookupByAlias looks up a container by alias, given the requesting container
+// and the network of the incoming request. It first does a container-scoped
+// alias search, followed by a network-scoped aliased search for alias. The
+// result is a collection of endpoints for the matching container that are
+// only on the same network as the requesting container.
+func lookupByAlias(ctx *network.Context, scope *network.Scope, reqc *network.Container, alias string) []*network.Endpoint {
+	// container specific alias search
+	cons := ctx.ContainersByAlias(fmt.Sprintf("%s:%s:%s", scope.Name(), reqc.Name(), alias))
+	if len(cons) == 0 {
+		// scope-wide alias search
+		cons = ctx.ContainersByAlias(fmt.Sprintf("%s:%s", scope.Name(), alias))
+	}
+
+	if len(cons) == 0 {
+		return nil
+	}
+
+	eps := make([]*network.Endpoint, len(cons))
+	for i, c := range cons {
+		eps[i] = c.Endpoint(scope)
+	}
+
+	return eps
+}
+
+// lookupByName looks up a container by name given the requesting container. It returns a collection
+// of endpoints on networks that both the requesting and matching container share.
+func lookupByName(ctx *network.Context, reqc *network.Container, name string) []*network.Endpoint {
+	var eps []*network.Endpoint
+	if m := ctx.Container(name); m != nil {
+		// look if networks overlap for the requesting container and the
+		// matched container
+		for _, ec := range reqc.Endpoints() {
+			if em := m.Endpoint(ec.Scope()); em != nil {
+				eps = append(eps, em)
```

Comment:
On line `340` you `make()` the slice up front.  Here you `append`.  Why the different idioms?